test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java

```diff
@@ -227,6 +228,32 @@ public void alertsIfPrivIpAndDropsPortNoInEvidence() throws HttpMalformedHeaderE
 			}
 		}
 	}
+	
+	@Test
+	public void shouldNotAlertIfPrivIpIsTarget() throws HttpMalformedHeaderException, URIException {
+		// Given
+		// ip and aws-hostname which get concatenated with the ports as candidates
+		String[] ipHost = new String[] {"10.0.2.2", "ip-10-0-2-2"};
+		String[] ports = new String[] {
+				":45876",
+				":8081",
+				":98", 
+				""
+		};
+		for (int pi = 0; pi < ports.length; pi++) {
+			for (int ii = 0; ii < ipHost.length; ii++) {
+				alertsRaised.clear();
+				String candidate = ipHost[ii] + ports[pi];
+				HttpMessage msg = createHttpMessage(candidate);
```

Comment:
It could use `createHttpMessage(String, String)`, instead of setting the URI again.